go/src/koding/klientctl/endpoint/machine/mount.go

```diff
@@ -0,0 +1,278 @@
+		return nil, err
+	}
+
+	// List mounts.
+	listMountReq := machinegroup.ListMountRequest{
+		ID:      machine.ID(options.ID),
+		MountID: mount.ID(options.MountID),
+	}
+	listMountRaw, err := k.Tell("machine.mount.list", listMountReq)
+	if err != nil {
+		return nil, err
+	}
+	listMountRes := machinegroup.ListMountResponse{}
+	if err := listMountRaw.Unmarshal(&listMountRes); err != nil {
+		return nil, err
+	}
+
+	return listMountRes.Mounts, nil
+}
+
+// UmountOptions stores options for `machine umount` call.
+type UmountOptions struct {
+	Identifier string // Mount identifier.
+	Log        logging.Logger
+}
+
+// Umount removes existing mount.
+func Umount(options *UmountOptions) (err error) {
+	if options == nil {
+		return errors.New("invalid nil options")
+	}
+
+	// TODO(ppknap): this is copied from klientctl old list and will be reworked.
+	k, err := klient.CreateKlientWithDefaultOpts()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error creating klient:", err)
+		return err
+	}
+
+	if err := k.Dial(); err != nil {
+		return err
+	}
+
+	// TODO: ask user to confirm unmounting.
+	fmt.Fprintf(os.Stdout, "Unmounting %s...\n", options.Identifier)
+
+	// Remove mount.
+	umountReq := machinegroup.UmountRequest{
+		Identifier: options.Identifier,
+	}
+	umountRaw, err := k.Tell("machine.umount", umountReq)
+	if err != nil {
+		return err
+	}
+	umountRes := machinegroup.UmountResponse{}
+	if err := umountRaw.Unmarshal(&umountRes); err != nil {
+		return err
+	}
+
+	fmt.Fprintf(os.Stdout, "Successfully unmounted %s (ID: %s)\n",
+		umountRes.Mount, umountRes.MountID)
+
+	return nil
+}
+
+// mountPointDirectory checks and prepares local directory for mounting.
+// Returned clean function can be used to remove resources in case of other
+// mounting errors.
+//
+// NOTE: This logic will be moved to klient.
+func mountPointDirectory(path string) (clean func(), err error) {
+	switch info, err := os.Stat(path); {
+	case os.IsNotExist(err):
+		// Create a new directory.
+		if err := os.MkdirAll(path, 0755); err != nil {
+			return nil, fmt.Errorf("cannot create destination directory: %s", err)
+		}
+		return func() {
+			os.RemoveAll(path)
+		}, nil
+	case err != nil:
+		return nil, fmt.Errorf("cannot stat destination directory: %s", err)
+	case !info.IsDir():
+		return nil, fmt.Errorf("file %q is not a directory", path)
+
+	}
+
+	// Provided directory already exists. Check if it's empty.
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open destination directory: %s", err)
+	}
+	defer f.Close()
+
+	switch _, err = f.Readdirnames(1); err {
+	case nil:
+		return nil, errors.New("destination directory is not empty")
+	case io.EOF:
+	default:
+		return nil, fmt.Errorf("destination directory error: %s", err)
+	}
+	clean = func() {
+		removeContent(path)
+	}
+
+	return clean, nil
+
+}
+
+// removeContent removes all files inside provided path but not the path itself.
+func removeContent(path string) error {
+	f, err := os.Open(path)
```

Comment:
yes, I want this path to exist, if the user mounts to non-existing directory and mounting fails, the created directory will be removed (L229). However, if the user mounts to existing directory I don't want to remove it on failure.

I could `os.RemoveAll` and `os.MkdirAll` but read [this](http://stackoverflow.com/questions/33450980/golang-remove-all-contents-of-a-directory) and decided not to(because of ensuring new directory permissions match old ones)